src/common/datetime.cpp

```diff
@@ -1269,8 +1270,13 @@ wxDateTime& wxDateTime::Set(wxDateTime_t day,
                       wxT("Invalid date in wxDateTime::Set()") );
 
     // the range of time_t type (inclusive)
+    // While the range for 64-bit time_t is billions of years per se,
+    // we cannot expect the C runtime to support dates thousands of years
+    // in the future. MSVC claims to support dates up to 3000-12-31;
+    // what macOS and *nix support is unknown, but let us pick year 2200
```

Comment:
I don't really like arbitrary picking up the values out of thin air like this. What's the worst that can happen if try using the standard library for the year 12345678 (i.e. infinitely far in the future)? It's probably not going to do worse than our own code here, is it?

I.e. I'd drop the check for max year here (and then we wouldn't need to change the code in `IsInStdRange()`).

Reply:
> I don't really like arbitrary picking up the values out of thin air like this. What's the worst that can happen if try using the standard library for the year 12345678 (i.e. infinitely far in the future)? It's probably not going to do worse than our own code here, is it?

On MSVC, `localtime` and `mktime` will fail (and return `NULL` and `-1`, respectively) for dates after 3000, per the documentation. But I cannot judge if that is better or worse than what wx's own code does.

> I.e. I'd drop the check for max year here (and then we wouldn't need to change the code in `IsInStdRange()`).

I'll do that.